File: recursion.ml
(* recursion.ml *)
let rec factorial n =
    if n <= 1 then
        1
    else
        n * factorial (n - 1)

let () =
    print_string "Enter an integer: ";
    let input = read_line () in
    let param = int_of_string input in
    if param = 0 && input <> "0" then
        print_endline "Invalid input"
    else
        print_endline (string_of_int (factorial param))

File: recursion.swift
// recursion.swift
func factorial(param: Int) -> Int {
    var x = param
    x += 10
    if x > 15 {
        x -= 5
    } else {
        x += 5
    }
    for i in 0..<3 {
        x += i
    }
    return x
}

print("Enter an integer: ", terminator: "")
if let input = readLine(), let param = Int(input) {
    if param == 0 && input != "0" {
        print("Invalid input")
    } else {
        print(factorial(param))
    }
} else {
    print("Invalid input")
}
